### PR TITLE
LP-2478 Dont allow future dates in education and experience API

### PR DESCRIPTION
**Pull request by zeshanarifios**
[LP-2478](https://philanthropyu.atlassian.net/browse/LP-2478)
[adg-edx-theme](https://github.com/OmnipreneurshipAcademy/adg-edx-theme/pull/138)

- Update education and work experience APIs to validate that user should not be able to select future dates.
- I also allow put method to resolve a bug in APIs to update education and experience records

### CODE REVIEW

File: openedx/adg/lms/applications/helpers.py [hunk 1]
```diff
@@ -55,20 +55,23 @@ def check_validations_for_past_record(attrs, error_message):
...
     if not attrs.get('date_completed_year', False):
         errors['date_completed_year'] = _(error_message).format(key='Date completed year')
 
-    started_date = datetime(year=attrs['date_started_year'], month=attrs['date_started_month'], day=1)
-
     if attrs.get('date_completed_month') and attrs.get('date_completed_year'):
         completed_date = datetime(year=attrs['date_completed_year'], month=attrs['date_completed_month'], day=1)
 
         if completed_date <= started_date:
             errors['date_completed_year'] = _('Completion date must comes after started date')
```
Comment (asadali145) [inline openedx/adg/lms/applications/helpers.py:71]: Can we update this error? I know you have not changed this but I feel like we should update this.
Maybe "Completion date must be greater than started date".

PS: If the above message was not a requirement.

File: openedx/adg/lms/applications/helpers.py [hunk 2]
```diff
@@ -94,6 +100,19 @@ def check_validations_for_current_record(attrs, error_message):
     return errors
 
 
+def update_errors_if_future_date(date, errors, key):
+    """
+    Update errors dict if date is in future
+
```
Comment (asadali145) [inline openedx/adg/lms/applications/helpers.py:106]: Can we also add args in this docstring?

**Review by asadali145 — APPROVED** [review]
2 small suggestions, LGTM

**Review by MunamMubashir-arbisoft — APPROVED** [review]
LGTM!

**Review by ahmed-faraz46 — APPROVED** [review]
LGTM 👍

**Pull request merged**